src/compare-tree/index.js

```diff
@@ -1,28 +1,31 @@
+/**
+ * @flow
+ * @private
+ */
+
 import Node from './node'
 import Query from './query'
-import { getKeyString } from '../key-path'
+import { getKeyString, type KeyPath } from '../key-path'
```

Comment:
KeyPath is a type alias for `string | Array<string>`. 

I believe the difference is that `type <module>` does not create a hard dependency, it only imports the type information. So if `getKeyString` weren't also imported here, the compare tree would not directly depend on the `key-path` module.